src/actions/importing/aioseo-posttype-defaults-settings-importing-action.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Yoast\WP\SEO\Actions\Importing;
+
+/**
+ * Importing action for AIOSEO posttype defaults settings data.
+ *
+ * @phpcs:disable Yoast.NamingConventions.ObjectNameDepth.MaxExceeded
+ */
+class Aioseo_Posttype_Defaults_Settings_Importing_Action extends Abstract_Aioseo_Settings_Importing_Action {
+
+	/**
+	 * The plugin of the action.
+	 */
+	const PLUGIN = 'aioseo';
+
+	/**
+	 * The type of the action.
+	 */
+	const TYPE = 'posttype_default_settings';
+
+	/**
+	 * The option_name of the AIOSEO option that contains the settings.
+	 */
+	const SOURCE_OPTION_NAME = 'aioseo_options_dynamic';
+
+	/**
+	 * The map of aioseo_options to yoast settings.
+	 *
+	 * @var array
+	 */
+	protected $aioseo_options_to_yoast_map = [];
+
+	/**
+	 * The tab of the aioseo settings we're working with.
+	 *
+	 * @var string
+	 */
+	protected $settings_tab = 'postTypes';
+
+	/**
+	 * Builds the mapping that ties AOISEO option keys with Yoast ones and their data transformation method.
+	 *
+	 * @return void
+	 */
+	protected function build_mapping() {
+		$post_type_objects = \get_post_types( [ 'public' => true ], 'objects' );
+
+		foreach ( $post_type_objects as $pt ) {
+			// Use all the custom post types that are public.
+			$this->aioseo_options_to_yoast_map[ '/' . $pt->name . '/title' ]           = [
+				'yoast_name'       => 'title-' . $pt->name,
+				'transform_method' => 'simple_import',
+			];
+			$this->aioseo_options_to_yoast_map[ '/' . $pt->name . '/metaDescription' ] = [
+				'yoast_name'       => 'metadesc-' . $pt->name,
+				'transform_method' => 'simple_import',
+			];
+
+			if ( $pt->name === 'attachment' ) {
+				$this->aioseo_options_to_yoast_map['/attachment/redirectAttachmentUrls'] = [
+					'yoast_name'       => 'disable-attachment',
+					'transform_method' => 'import_redirect_attachment',
+				];
+			}
+		}
+	}
+
+	/**
+	 * Transforms the redirect_attachment setting.
+	 *
+	 * @param string $redirect_attachment The redirect_attachment setting.
+	 *
+	 * @return string The transformed redirect_attachment setting.
+	 */
+	public function import_redirect_attachment( $redirect_attachment ) {
+		switch ( $redirect_attachment ) {
+			case 'disabled':
+				return false;
+
+			case 'attachment':
+			case 'attachment_parent':
+				return true;
+		}
```

Comment:
```suggestion
		}
		return false;
```
an extra safeguard against crap data perhaps?

Reply:
Makes perfect sense 👌 

I'll `return true` in those cases though, considering this is our recommended setting for that :)